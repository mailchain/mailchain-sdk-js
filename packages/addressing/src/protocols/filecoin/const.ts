import { SECP256K1PrivateKey } from '@mailchain/crypto';
import { decodeHex } from '@mailchain/encoding';

// Alice
export const AliceFilStr = 'f410faiikgixcdd7alrsbpunjlen4etkcarbfbghn7wi';
export const AliceFilEthAddressStr = '0x0210a322e218fe05c6417d1a9591bc24d4204425';
export const AliceFilEthPrivateKeyStr = '6c115551accbd74b46401a531af1dd0890034c18137dddab52a53c626e75b051';
export const AliceFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(AliceFilEthPrivateKeyStr));

// Bob
export const BobFilAddressStr = 'f410f5blfdi7tmto6o2aa2up4vwtdm7qcnaiqpeazuwy';
export const BobFilEthAddressStr = '0xe85651a3f364dde76800d51fcada6367e0268110';
export const BobFilEthPrivateKeyStr = '2c6b29d5545e5804b7fb7e65404782e57dddea84cbcec9b2586add6403808965';
export const BobFilEthPrivateKey = new SECP256K1PrivateKey(decodeHex(BobFilEthPrivateKeyStr));

export enum FilPrefix {
	F4ETHEREUM = 'f410',
	T4ETHEREUM = 't410',
}
