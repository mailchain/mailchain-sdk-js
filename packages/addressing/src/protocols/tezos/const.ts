export enum Prefix {
	TZ1 = 'tz1',
	TZ2 = 'tz2',
	TZ3 = 'tz3',
	KT = 'KT',
	KT1 = 'KT1',

	EDSK2 = 'edsk2',
	SPSK = 'spsk',
	P2SK = 'p2sk',

	EDPK = 'edpk',
	SPPK = 'sppk',
	P2PK = 'p2pk',

	EDESK = 'edesk',
	SPESK = 'spesk',
	P2ESK = 'p2esk',

	EDSK = 'edsk',
	EDSIG = 'edsig',
	SPSIG = 'spsig',
	P2SIG = 'p2sig',
	SIG = 'sig',
	EXPR = 'expr',
}

export const prefix = {
	[Prefix.TZ1]: new Uint8Array([6, 161, 159]),
	[Prefix.TZ2]: new Uint8Array([6, 161, 161]),
	[Prefix.TZ3]: new Uint8Array([6, 161, 164]),
	[Prefix.KT]: new Uint8Array([2, 90, 121]),
	[Prefix.KT1]: new Uint8Array([2, 90, 121]),

	[Prefix.EDSK]: new Uint8Array([43, 246, 78, 7]),
	[Prefix.EDSK2]: new Uint8Array([13, 15, 58, 7]),
	[Prefix.SPSK]: new Uint8Array([17, 162, 224, 201]),
	[Prefix.P2SK]: new Uint8Array([16, 81, 238, 189]),

	[Prefix.EDPK]: new Uint8Array([13, 15, 37, 217]),
	[Prefix.SPPK]: new Uint8Array([3, 254, 226, 86]),
	[Prefix.P2PK]: new Uint8Array([3, 178, 139, 127]),

	[Prefix.EDESK]: new Uint8Array([7, 90, 60, 179, 41]),
	[Prefix.SPESK]: new Uint8Array([0x09, 0xed, 0xf1, 0xae, 0x96]),
	[Prefix.P2ESK]: new Uint8Array([0x09, 0x30, 0x39, 0x73, 0xab]),

	[Prefix.EDSIG]: new Uint8Array([9, 245, 205, 134, 18]),
	[Prefix.SPSIG]: new Uint8Array([13, 115, 101, 19, 63]),
	[Prefix.P2SIG]: new Uint8Array([54, 240, 44, 52]),
	[Prefix.SIG]: new Uint8Array([4, 130, 43]),
	[Prefix.EXPR]: new Uint8Array([13, 44, 64, 27]),
};
