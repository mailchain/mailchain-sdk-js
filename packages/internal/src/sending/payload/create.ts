import { KindNaClSecretKey, SignerWithPublicKey } from '@mailchain/crypto';
import { EncodingTypes } from '@mailchain/encoding';
import isArrayBuffer from 'lodash/isArrayBuffer';
import { PayloadHeaders } from '../../transport/payload/headers';
import { ContentType, Payload } from '../../transport';

export async function createPayload<C extends ContentType>(
	signerMessagingKey: SignerWithPublicKey,
	content: Buffer | Uint8Array,
	contentType: C,
): Promise<Payload<{ ContentType: C } & PayloadHeaders>> {
	return {
		Headers: {
			Origin: signerMessagingKey.publicKey,
			ContentSignature: await signerMessagingKey.sign(content),
			Created: new Date(),
			ContentLength: content.length,
			ContentType: contentType,
			ContentEncoding: EncodingTypes.Base64,
			ContentEncryption: KindNaClSecretKey,
		},
		Content: isArrayBuffer(content) ? Buffer.from(content) : content,
	};
}
