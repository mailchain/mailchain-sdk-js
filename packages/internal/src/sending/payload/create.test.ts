import { AliceED25519PrivateKey, BobED25519PrivateKey } from '@mailchain/crypto/ed25519/test.const';
import { createPayload } from './create';

describe('createPayload', () => {
	const content = Buffer.from('content');

	beforeAll(() => {
		jest.useFakeTimers().setSystemTime(new Date('2022-06-06'));
	});
	afterAll(() => {
		jest.useRealTimers();
	});
	it('should create a payload - alice key', async () => {
		expect(
			await createPayload(
				AliceED25519PrivateKey,
				content,
				'application/vnd.mailchain.verified-credential-request',
				{ plugin: { value: 'value' } },
			),
		).toMatchSnapshot();
	});

	it('should create a payload - bob key', async () => {
		expect(
			await createPayload(
				BobED25519PrivateKey,
				content,
				'application/vnd.mailchain.verified-credential-request',
				{ plugin: { value: 'value' } },
			),
		).toMatchSnapshot();
	});
});
