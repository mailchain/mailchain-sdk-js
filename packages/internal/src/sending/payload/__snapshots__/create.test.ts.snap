// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPayload should create a payload - alice key 1`] = `
{
  "Content": {
    "data": [
      99,
      111,
      110,
      116,
      101,
      110,
      116,
    ],
    "type": "Buffer",
  },
  "Headers": {
    "ContentEncoding": "base64/plain",
    "ContentEncryption": "nacl-secret-key",
    "ContentLength": 7,
    "ContentSignature": Uint8Array [
      170,
      222,
      217,
      108,
      230,
      127,
      31,
      43,
      143,
      207,
      132,
      133,
      102,
      247,
      30,
      16,
      219,
      109,
      244,
      166,
      131,
      75,
      135,
      159,
      226,
      228,
      79,
      115,
      6,
      207,
      195,
      173,
      144,
      150,
      106,
      188,
      120,
      246,
      75,
      68,
      40,
      81,
      198,
      111,
      3,
      9,
      111,
      112,
      59,
      95,
      29,
      231,
      42,
      234,
      142,
      169,
      117,
      48,
      248,
      247,
      179,
      136,
      2,
      10,
    ],
    "ContentType": "application/vnd.mailchain.verified-credential-request",
    "Created": 2022-06-06T00:00:00.000Z,
    "Origin": ED25519PublicKey {
      "bytes": Uint8Array [
        114,
        60,
        170,
        35,
        165,
        181,
        17,
        175,
        90,
        215,
        183,
        239,
        96,
        118,
        228,
        20,
        171,
        126,
        117,
        169,
        220,
        145,
        14,
        166,
        14,
        65,
        122,
        43,
        119,
        10,
        86,
        113,
      ],
      "curve": "ed25519",
    },
    "PluginHeaders": {
      "plugin": {
        "value": "value",
      },
    },
  },
}
`;

exports[`createPayload should create a payload - bob key 1`] = `
{
  "Content": {
    "data": [
      99,
      111,
      110,
      116,
      101,
      110,
      116,
    ],
    "type": "Buffer",
  },
  "Headers": {
    "ContentEncoding": "base64/plain",
    "ContentEncryption": "nacl-secret-key",
    "ContentLength": 7,
    "ContentSignature": Uint8Array [
      162,
      180,
      3,
      119,
      144,
      21,
      189,
      245,
      208,
      155,
      140,
      87,
      185,
      191,
      149,
      46,
      47,
      111,
      159,
      238,
      112,
      56,
      3,
      27,
      53,
      187,
      252,
      204,
      127,
      18,
      108,
      171,
      91,
      167,
      162,
      73,
      99,
      185,
      178,
      213,
      12,
      153,
      149,
      187,
      219,
      169,
      150,
      150,
      150,
      242,
      175,
      54,
      233,
      207,
      169,
      18,
      34,
      39,
      220,
      79,
      97,
      147,
      134,
      2,
    ],
    "ContentType": "application/vnd.mailchain.verified-credential-request",
    "Created": 2022-06-06T00:00:00.000Z,
    "Origin": ED25519PublicKey {
      "bytes": Uint8Array [
        46,
        50,
        47,
        135,
        64,
        198,
        1,
        114,
        17,
        26,
        200,
        234,
        220,
        221,
        162,
        81,
        47,
        144,
        208,
        109,
        14,
        80,
        62,
        241,
        137,
        151,
        154,
        21,
        155,
        236,
        225,
        232,
      ],
      "curve": "ed25519",
    },
    "PluginHeaders": {
      "plugin": {
        "value": "value",
      },
    },
  },
}
`;
